Libraries/LibJS/Runtime/Shape.h

```diff
@@ -84,34 +84,34 @@ class Shape final : public Cell {
     Object* prototype() { return m_prototype; }
     const Object* prototype() const { return m_prototype; }
 
-    Optional<PropertyMetadata> lookup(const FlyString&) const;
-    const HashMap<FlyString, PropertyMetadata>& property_table() const;
+    Optional<PropertyMetadata> lookup(const PropertyName&) const;
+    const HashMap<PropertyName, PropertyMetadata>& property_table() const;
     size_t property_count() const;
 
     struct Property {
-        FlyString key;
+        PropertyName key;
```

Comment:
`PropertyName` is a large object (12 bytes compared to `FlyString` being 4 bytes) that was never meant for persistent storage like this. In this context, the numeric index part of `PropertyName` is completely dead weight since shapes are unconcerned with indexes. The type field of `PropertyName` is also way too huge and really only needs a single bit.

Let's instead find a way to store either a symbol or a string here. Perhaps we could have something like a `SymbolOrString` class that is internally a pointer to a `StringImpl` (interned through `FlyString`) and using the lowest bit of the pointer to indicate whether it's a string or a symbol?